dmaws/build.py

```diff
@@ -21,8 +23,22 @@ def run_git_cmd(args, cwd, stdout=None):
     )
 
 
-def get_application_name(cwd):
-    repo_url = get_repo_url(cwd)
+def clone_or_update(repo_url):
+    app_name = get_application_name_from_url(repo_url)
+    repository_path = os.path.join(REPOS_PATH,
+                                   'digitalmarketplace-{}'.format(app_name))
+    if not os.path.exists(REPOS_PATH):
+        os.mkdir(REPOS_PATH)
+    if not os.path.exists(repository_path):
+        run_git_cmd(['clone', repo_url], REPOS_PATH)
+    else:
+        run_git_cmd(['reset', '--hard', 'origin'], repository_path)
```

Comment:
We should remove all untracked and ignored files at this point too, including build artefacts from previous builds and npm/bower modules